source/test/F0.Minesweeper.Logic.Tests/LocationShuffler/FisherYatesLocationShufflerTest.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.Collections.Generic;
+using F0.Minesweeper.Logic.Abstractions;
+using F0.Minesweeper.Logic.LocationShuffler;
+using FluentAssertions;
+using Xunit;
+
+namespace F0.Minesweeper.Logic.Tests.LocationShuffler
+{
+	public class FisherYatesLocationShufflerTest
+	{
+		[Theory]
+		[InlineData(1)]
+		[InlineData(2)]
+		[InlineData(3)]
+		[InlineData(4)]
+		[InlineData(5)]
+		[InlineData(10)]
+		[InlineData(13)]
+		[InlineData(25)]
+		public void ShuffleAndTake_WithCount_DoesReturnCorrectCount(int count)
+		{
+			ILocationShuffler locationShufflerUnderTest = new FisherYatesLocationShuffler();
+			IReadOnlyCollection<Location> resultingLocations = locationShufflerUnderTest.ShuffleAndTake(field, count);
+
+			resultingLocations.Should().HaveCount(count);
+		}
+
+		[Fact]
+		public void ShuffleAndTake_WithCountGreaterThanField_ThrowsArgumentOutOfRangeException()
+		{
+			ILocationShuffler locationShufflerUnderTest = new FisherYatesLocationShuffler();
+			Action shuffleAction = () => locationShufflerUnderTest.ShuffleAndTake(field, field.Length + 1);
+
+			shuffleAction.Should().Throw<ArgumentOutOfRangeException>();
+		}
+
+		[Theory]
+		[MemberData(nameof(FisherYatesLocationShufflerTestData.TestData), MemberType = typeof(FisherYatesLocationShufflerTestData))]
```

Comment:
I'd remove the intermediate `class`, elide the `MemberType`, and just `nameof` the "Test-Data"-Method directly.